benches/lua.rs

```diff
@@ -9,167 +8,121 @@ use vector::{
     Event,
 };
 
-fn add_fields(c: &mut Criterion) {
+fn bench_add_fields(c: &mut Criterion) {
+    let event = Event::new_empty_log();
+
     let key = "the key";
     let value = "this is the value";
 
-    let value_bytes_native = Bytes::from(value).into();
-    let value_bytes_v1 = Bytes::from(value).into();
-    let value_bytes_v2 = Bytes::from(value).into();
+    let mut group = c.benchmark_group("lua_add_fields");
+    group.throughput(Throughput::Elements(1));
 
-    c.bench(
-        "lua_add_fields",
-        Benchmark::new("native", move |b| {
-            b.iter_with_setup(
```

Comment:
Switched away from `b.iter_with_setup` as:

* It's soft deprecated
* It was running this once per iteration which seemed unnecessary and possibly working against the purpose of the benchmark to measure the throughput